### PR TITLE
dev-cmd/audit: add libressl to uses_from_macos allow list

### PR DESCRIPTION
**Pull request by SeekingMeaning**
- [ ] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Required for Homebrew/homebrew-core#58190

### CODE REVIEW

**Comment by SMillerDev** [comment]
@Bo98 mentioned we can't actually build against it. So this wouldn't be valid.

**Comment by SeekingMeaning** [comment]
It's technically called `USES_FROM_MACOS_ALLOWLIST` but personally I'd call it `USES_FROM_MACOS_WHITELIST` since it specifies which formulae are okay to use `depends_on` (or at least that's my understanding)

This is where `USES_FROM_MACOS_ALLOWLIST` is used: https://github.com/Homebrew/brew/blob/master/Library/Homebrew/dev-cmd/audit.rb#L382-L391

**Comment by SMillerDev** [comment]
It does, but we can't depend on that libressl since it doesn't expose any headers or anything AFAIK.

**Comment by MikeMcQuaid** [comment]
> It's technically called `USES_FROM_MACOS_ALLOWLIST` but personally I'd call it `USES_FROM_MACOS_WHITELIST` since it specifies which formulae are okay to use `depends_on` (or at least that's my understanding)

We are intentionally using the term "allow list" instead of "whitelist". 

@SMillerDev's point here is valid but I'm not entirely sure I understand what problem this fixes.

**Comment by SeekingMeaning** [comment]
> It does, but we can't depend on that libressl since it doesn't expose any headers or anything AFAIK.

I'm not sure I understand. Are you referring to macOS libressl or Homebrew libressl?

> @SMillerDev's point here is valid but I'm not entirely sure I understand what problem this fixes.

This allows new formulae to `depends_on "libressl"`. In `brew`'s current state, here is the output for `brew audit --new-formula`:

```console
% brew audit --new-formula curl-libressl
curl-libressl:
  * Dependency 'libressl' is provided by macOS; please replace 'depends_on' with 'uses_from_macos'.
Error: 1 problem in 1 formula detected
```

**Comment by MikeMcQuaid** [comment]
> Are you referring to macOS libressl or Homebrew libressl?

macOS.



> This error doesn't make sense because we can't compile with Apple's builtin libressl.


Fine with me for adding it for that reason alone. @Bo98 and @SMillerDev any objections?

**Comment by SMillerDev** [comment]
> This error doesn't make sense because we can't compile with Apple's builtin libressl.

I don't know if we should claim that it's provided by macOS then.

**Comment by MikeMcQuaid** [comment]
> I don't know if we should claim that it's provided by macOS then.

It is provided by macOS. This is to stop an audit message that doesn't make sense (and, as far as I can tell, only that message).

**Comment by Bo98** [comment]
The name `USES_FROM_MACOS_ALLOWLIST` is indeed a bit misleading here. Makes it seem like the other `uses_from_macos` audit at first glance when it’s actually about allowing `depends_on` on `provided_by_macos` formulae.

File: Library/Homebrew/dev-cmd/audit.rb [hunk 1]
```diff
@@ -323,6 +323,7 @@ def audit_formula_name
     USES_FROM_MACOS_ALLOWLIST = %w[
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/audit.rb:323]:
```suggestion
    PROVIDED_BY_MACOS_DEPENDS_ON_ALLOWLIST = %w[
```

**Comment by MikeMcQuaid** [comment]
Thanks @SeekingMeaning!

**Pull request merged**